ProfilerGUI/Source/Configurator/UploadViewModel.cs

```diff
@@ -0,0 +1,193 @@
+                        break;
+
+                    case 2:
+                        if (Config == null)
+                        {
+                            Config = new UploadConfig();
+                        }
+                        Config.Teamscale = null;
+                        Config.Directory = string.Empty;
+                        break;
+                }
+
+                // force all UI elements to be refreshed since this affects almost all of them
+                PropertyChanged.Raise(this, null);
+            }
+        }
+
+        /// <summary>
+        /// The directory to upload traces to.
+        /// </summary>
+        public string Directory
+        {
+            get => Config?.Directory;
+            set
+            {
+                Config.Directory = value;
+                PropertyChanged.Raise(this);
+            }
+        }
+
+        /// <summary>
+        /// Whether the configuration UI for the Teamscale server should be shown.
+        /// </summary>
+        public bool IsTeamscaleConfigVisible { get => Config != null && Config.Teamscale != null; }
+
+        /// <summary>
+        /// Whether the configuration UI for the directory to copy to should be shown.
+        /// </summary>
+        public bool IsDirectoryConfigVisible { get => Config != null && Config.Teamscale == null; }
+
+        /// <summary>
+        /// Whether the configuration UI for the version assembly should be shown.
+        /// </summary>
+        public bool IsVersionAssemblyVisible { get => Config != null; }
+
+        private ValidationResult internalValidationResult = null;
+
+        /// <summary>
+        /// The general error message.
+        /// </summary>
+        public ValidationResult ValidationResult
+        {
+            get => internalValidationResult;
+            set
+            {
+                internalValidationResult = value;
+                PropertyChanged.Raise(this);
+            }
+        }
+
+        public UploadViewModel(UploadConfig config)
+        {
+            Config = config;
+        }
+
+        private void ShowErrorMessage(String message)
+        {
+            ValidationResult = new ValidationResult(false, message);
+        }
+
+        /// <summary>
+        /// Checks if current config is valid.
+        /// Shows an error message in the UI if not.
+        /// </summary>
+        public async Task<bool> Validate()
+        {
+            IEnumerable<string> errors = Config?.Validate();
+            if (errors != null && errors.Any())
+            {
+                ShowErrorMessage(string.Join("\r\n", errors));
+                return false;
+            }
+
+            return await CheckTeamscaleConnection();
+        }
+
+        /// <summary>
+        /// Checks if the connection to Teamscale can be established.
+        /// Shows an error message in the UI if not.
+        /// </summary>
+        private async Task<bool> CheckTeamscaleConnection()
```

Comment:
We're now duplicating functionality from the teamscale client lib. I think it's a good time to integrate it now and add features there if needed.

Reply:
created #30 for this. As far as I remember from our last discussion, we'd first need to turn that into a nuget package and then still have to rewrite this code here, push some stuff to the lib and retest the entire thing. I'd rather get this merged soon so I can use this code in the next pilot and do the maintenance changes afterwards without the time pressure